### PR TITLE
ti_crowdstrike: improve error reporting for unsuccessful API requests

### LINKED ISSUE #9051: ti_crowdstrike: authentication failure errors are not helpfully logged
When an attempt to connect to the https://api.crowdstrike.com /iocs/combined/indicator/v1 endpoint fails due to an authentication failure, we have the information required to be able to log a helpful message into the elastic-agent logs since we can conditionally return an error message based on the response status code.

```
{
    "log.level": "debug",
    "@timestamp": "2024-02-02T09:55:40.998Z",
    "message": "HTTP response",
    "transaction.id": "01G1HAVDTTQ12-1",
    "http.response.status_code": 403,
    "http.response.body.content": "{\n \"meta\": {\n  \"query_time\": 1.3e-7,\n  \"powered_by\": \"crowdstrike-api-gateway\",\n  \"trace_id\": \"c2d3b314-2cdb-466d-963e-9e1ef4f1152f\"\n },\n \"errors\": [\n  {\n   \"code\": 403,\n   \"message\": \"access denied, authorization failed\"\n  }\n ]\n}",
    "http.response.body.bytes": 231,
    "http.response.mime_type": "application/json",
    "event.original": "HTTP/1.1 403 Forbidden\r\nConnection: close\r\nContent-Length: 230\r\nContent-Type: application/json\r\nDate: Fri, 2 Feb 2024 9:55:40 GMT\r\nServer: nginx\r\nStrict-Transport-Security: max-age=31536000; includeSubDomains\r\nX-Content-Type-Options: nosniff\r\nX-Cs-Traceid: c2d3b314-2cdb-466d-963e-9e1ef4f1152f\r\nX-Ratelimit-Limit: 25\r\nX-Ratelimit-Remaining: 24\r\n\r\n",
    "ecs.version": "1.6.0"
}
```

What happens instead is that we attempt to [continue the evaluation](https://github.com/elastic/integrations/blob/ee77be723403d225811b7dc7ecaae179988811a3/packages/ti_crowdstrike/data_stream/ioc/agent/stream/cel.yml.hbs#L51). Resulting in an uninformative error being logged.
```
{
    "log.level": "error",
    "@timestamp": "2024-02-02T09:55:40.998Z",
    "message": "failed evaluation",
    "component": {
        "binary": "filebeat",
        "dataset": "elastic_agent.filebeat",
        "id": "cel-388088e4-286d-4311-8aab-5cba979863f3",
        "type": "cel"
    },
    "log": {
        "source": "cel-388088e4-286d-4311-8aab-5cba979863f3"
    },
    "service.name": "filebeat",
    "id": "cel-ti_crowdstrike.ioc-ec453c6f-df6b-4a6c-9a43-834b8eec4276",
    "log.logger": "input.cel",
    "ecs.version": "1.6.0",
    "log.origin": {
        "file.line": 243,
        "file.name": "cel/input.go"
    },
    "input_source": "https://api.crowdstrike.com",
    "input_url": "https://api.crowdstrike.com",
    "error": {
        "message": "failed eval: no such key: resources"
    }
}
```

We should conditionally evaluate and log the body as an error message if the status is non-200.

The intel data stream has a [similar defect](https://github.com/elastic/integrations/blob/ee77be723403d225811b7dc7ecaae179988811a3/packages/ti_crowdstrike/data_stream/intel/agent/stream/cel.yml.hbs#L51).

### PR DESCRIPTION
**Pull request by efd6**
## Proposed commit message

Previously, failed requests would continue through the standard happy path, potentially resulting in an uninformative CEL error when the data shape does not match the program's expectations. We have the HTTP status code, so make use of this to return an error object in the case that we get a non-success.

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ] 

## Related issues

- Closes #9051

### CODE REVIEW

File: packages/ti_crowdstrike/data_stream/intel/agent/stream/cel.yml.hbs [hunk 1]
```diff
@@ -47,48 +47,66 @@ program: |
...
-              :
-                state.cursor.last_timestamp
-            )
-          :
-            (now - duration(state.initial_interval)).format(time_layout.RFC3339)
-        ),
-      },
-  }))
+  ).do_request().as(resp, 
+    resp.StatusCode == 200 ?
```
Comment (efd6) [inline packages/ti_crowdstrike/data_stream/intel/agent/stream/cel.yml.hbs:51]: Query whether we know for sure that Crowdstrike always returns a 200 status for success.

**Review by chrisberkhout — APPROVED** [review]
Looks good if it is always 200 for success. If we can't confirm, I think 200 would be an okay assumption and 2XX would be a good assumption.

**Comment by NateUT99** [comment]
Customer (and CrowdStrike user) here... per their API specification swagger page, it will always return a 200. This corresponds to what I have seen in my testing as well. I hope this is helpful!

**Comment by efd6** [comment]
@NateUT99 Thank you, this is very helpful.

**Pull request merged**